src/main/java/com/example/bookstore/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,126 @@
+package com.example.bookstore.service.order;
+
+import com.example.bookstore.dto.order.OrderDto;
+import com.example.bookstore.dto.order.OrderItemDto;
+import com.example.bookstore.dto.order.PlaceOrderRequestDto;
+import com.example.bookstore.dto.order.UpdateStatusRequestDto;
+import com.example.bookstore.exception.EntityNotFoundException;
+import com.example.bookstore.exception.InvalidStatusTransitionException;
+import com.example.bookstore.mapper.OrderItemMapper;
+import com.example.bookstore.mapper.OrderMapper;
+import com.example.bookstore.model.CartItem;
+import com.example.bookstore.model.Order;
+import com.example.bookstore.model.Order.OrderStatus;
+import com.example.bookstore.model.OrderItem;
+import com.example.bookstore.model.ShoppingCart;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.order.OrderRepository;
+import com.example.bookstore.repository.order.item.OrderItemRepository;
+import com.example.bookstore.repository.shopping.cart.CartItemRepository;
+import com.example.bookstore.repository.shopping.cart.ShoppingCartRepository;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class OrderServiceImpl implements OrderService {
+    private static final Map<OrderStatus, Set<OrderStatus>> VALID_TRANSITIONS = Map.of(
+            OrderStatus.PENDING, Set.of(OrderStatus.IN_PROGRESS, OrderStatus.CANCELED),
+            OrderStatus.IN_PROGRESS, Set.of(OrderStatus.SHIPPED, OrderStatus.CANCELED),
+            OrderStatus.SHIPPED, Set.of(OrderStatus.DELIVERED, OrderStatus.CANCELED)
+    );
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderRepository orderRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemRepository orderItemRepository;
+
+    @Override
+    public OrderDto placeOrder(User user, PlaceOrderRequestDto requestDto) {
+        ShoppingCart userCart = shoppingCartRepository.findByUserId(user.getId());
+        Set<CartItem> cartItems = userCart.getCartItems();
+        if (cartItems.isEmpty()) {
+            throw new EntityNotFoundException("Shopping Cart can't be empty");
```

Comment:
provide parameter to message
```suggestion
            throw new OrderProcessingException("Shopping Cart can't be empty");
```